lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -91,6 +94,45 @@ sub view ($self) {
     $self->viewimg;
 }
 
+sub _checkout_needledir_or_default {
+    my $job = shift;
+    my $tmp_needles_dir = $job->needle_dir;
+    my $real_needle_dir = realpath($tmp_needles_dir) // $tmp_needles_dir;
```

Comment:
`return $tmp_needles_dir unless my $casedir = $job->settings->single({key => 'CASEDIR'});` would save us one level of indentation.